integrations/clickup/clickup/client.py

```diff
@@ -0,0 +1,186 @@
+
+    async def get_clickup_teams(self) -> AsyncGenerator[List[dict[str, Any]], None]:
+        """Get all Clickup teams."""
+        url = f"{self.api_url}/team"
+        response = await self._send_api_request(url)
+        teams = response.get("teams", [])
+        yield teams
+
+    async def _get_spaces_in_team(self, team_id: str) -> AsyncGenerator[str, None]:
+        """Get all spaces in a workspace."""
+        url = f"{self.api_url}/team/{team_id}/space"
+        params = {"archived": "false"}
+        response = await self._send_api_request(url, params)
+        spaces = response.get("spaces")
+        for space in spaces:
+            yield space.get("id")
+
+    async def _get_folders_in_space(self, team_id: str) -> AsyncGenerator[str, None]:
+        """Get all folders in a space."""
+        async for space in self._get_spaces_in_team(team_id):
+            url = f"{self.api_url}/space/{space}/folder"
+            params = {"archived": "false"}
+            response = await self._send_api_request(url, params)
+            folders = response.get("folders")
+            for folder in folders:
+                yield folder.get("id")
+
+    async def get_folder_projects(self) -> AsyncGenerator[List[dict[str, Any]], None]:
+        """Get all projects with a folder parent."""
+        for team_id in self.team_key:
+            async for folder_id in self._get_folders_in_space(team_id):
+                url = f"{self.api_url}/folder/{folder_id}/list"
+                params = {"archived": "false"}
+                response = await self._send_api_request(url, params)
+                projects = response.get("lists")
+                yield [
+                    {**project, CustomProperties.TEAM_ID: team_id}
+                    for project in projects
+                ]
+
+    async def get_folderless_projects(
+        self,
+    ) -> AsyncGenerator[List[dict[str, Any]], None]:
+        """Get all projects without a folder parent."""
+        for team_id in self.team_key:
+            async for space_id in self._get_spaces_in_team(team_id):
+                url = f"{self.api_url}/space/{space_id}/list"
+                params = {"archived": "false"}
+                response = await self._send_api_request(url, params)
+                projects = response.get("lists")
+                yield [
+                    {**project, CustomProperties.TEAM_ID: team_id}
+                    for project in projects
+                ]
+
+    async def get_single_project(self, list_id: str) -> dict[str, Any]:
+        """Get a single project by folder_id."""
+        url = f"{self.api_url}/list/{list_id}"
+        response = await self._send_api_request(url)
+        return response
+
+    async def get_paginated_issues(self) -> AsyncGenerator[List[dict[str, Any]], None]:
+        """Get all issues in a project."""
+        if self.team_key is None:
+            await self.initialize_team_key()
+        for team_id in self.team_key:
```

Comment:
Avoid fetching resources by relying on the self.team_key

Reply:
I have refactored to account for this